src/main/java/com/example/spring_bank_project/shared/domain/factory/UuidFactory.java

```diff
@@ -0,0 +1,12 @@
+package com.example.spring_bank_project.shared.domain.factory;
+
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class UuidFactory {
+    public UUID newUuid() {
+        return UUID.randomUUID();
```

Comment:
imo mocna przesada z takim factory, jakie wady ma po prostu wywoalnie statica UUID.randomUUID()? bedziesz mockowal proces generowania uuid? ppo co

Reply:
Bo w przyszłości będę pisał do tego testy. I to factory bedę mógł zastąpić szpiegiem, i w przypadkach generowania randomowych idków będę wiedział po czym wyciągnąć z bazki

Reply:
mozesz zawsze po prostu sprawdzic id pod ktorym sie zapisalo wywolujac var id = repository.save(entity).getId(), spy na generowaniu id brzmi jak straszny overkill

Reply:
imho niekoniecznie. Miałem kiedyś taki case że zapisywałem do bazki naraz 20 obiektów i chciałem w teście sprawdzić czy się zapisały do bazki. I asercja wyglądała tak
```
public function assertSavedToDatabase(UuidFactorySpy $spy): void
{
 $expectedIds = $spy->getIds();

foreach($expectedIds as $expectedId)
{
$object = $this->entityManager->find(Entity::class, $expectedId);
assertNotNull($object);
}
}
```